[docs] Revise the related projects page (#32180)
mui/material-ui: docs/data/material/components/dialogs/dialogs.md

Follow the [Modal limitations section](/material-ui/react-modal/#limitations).

## Complementary projects

### Material UI Confirm

![stars](https://img.shields.io/github/stars/jonatanklosko/material-ui-confirm)
![npm downloads](https://img.shields.io/npm/dm/material-ui-confirm.svg)

> This package provides easy to use confirmation dialogs to simplify confirming user actions without writing boilerplate code.

Reviewer: ```suggestion
This package provides dialogs for confirming user actions without writing boilerplate code.
```
